src/term.rs

```diff
@@ -912,6 +927,17 @@ impl RichTerm {
                     pos,
                 }
             }
+            Term::LetBlock(defs, rest) => {
+                let mut data = vec![];
+                for (id, term) in defs.iter() {
+                    data.push((id.clone(), term.clone().traverse(f, state, method)?));
+                }
```

Comment:
Same remark about cloning. Also just for the sake of mentioning it, you can use `collect` on a `Iterator<Item = Result<_>>` to get a `Result<Vec<_>, _>`, something like:

```
let data : Result<Vec<_>, _> = defs
    .into_iter()
    .map(|id, t| Ok((id, t.traverse(f, state, method)?)))
    .collect();
...
                     term: Box::new(Term::LetBlock(data?, rest)),
```

I'm not saying you should necessarily change it (it's a matter of taste in the end), but just FYI.